### DISCUSSION TITLE
Default Encoding scheme

### DISCUSSION THREAD

**Discussion by RFalowo** [fangwei123456/SpikingFlow#521] [Q&A]
What is the default encoding scheme of SNN input layers and how could you change it?

**Comment by fangwei123456** [comment]
Hi, in previous research, many ariticles used the poisson encoder. But now the directly encoding method is more popoular because the accuracy loss is less that other encoders.


> Because event-based benchmark datasets are rare ([Hu et al., 2016](https://www.frontiersin.org/journals/neuroscience/articles/10.3389/fnins.2017.00682/full#B16); [Rueckauer and Delbruck, 2016](https://www.frontiersin.org/journals/neuroscience/articles/10.3389/fnins.2017.00682/full#B38)), conventional frame-based image databases such as MNIST ([LeCun et al., 1998](https://www.frontiersin.org/journals/neuroscience/articles/10.3389/fnins.2017.00682/full#B22)) or CIFAR ([Krizhevsky, 2009](https://www.frontiersin.org/journals/neuroscience/articles/10.3389/fnins.2017.00682/full#B20)) have been used to evaluate the classification error rate of the converted SNN. Previous methods ([Cao et al., 2015](https://www.frontiersin.org/journals/neuroscience/articles/10.3389/fnins.2017.00682/full#B2); [Diehl et al., 2015](https://www.frontiersin.org/journals/neuroscience/articles/10.3389/fnins.2017.00682/full#B8)) usually transform the analog input activations, e.g., gray levels or RGB values, into Poisson firing rates. But this transformation introduces variability into the firing of the network and impairs its performance. Here, we interpret the analog input activations as constant currents.

(cited from https://www.frontiersin.org/journals/neuroscience/articles/10.3389/fnins.2017.00682/full )



> What is the default encoding scheme of SNN

The default case is no encoder.
